src/utils/spo.ts

```diff
@@ -1485,6 +1488,122 @@ export const spo = {
     await request.post(requestOptions);
   },
 
+  async getFileAsListItemByUrl(webUrl: string, url: string, logger?: Logger, verbose?: boolean): Promise<any> {
+    if (verbose && logger) {
+      logger.logToStderr(`Getting the file properties with url ${url}`);
+    }
+
+    const serverRelativePath = urlUtil.getServerRelativePath(webUrl, url);
+    const requestUrl = `${webUrl}/_api/web/GetFileByServerRelativePath(DecodedUrl=@f)?$expand=ListItemAllFields&@f='${formatting.encodeQueryParameter(serverRelativePath)}'`;
+
+    const requestOptions: CliRequestOptions = {
+      url: requestUrl,
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    const file = await request.get<FileProperties>(requestOptions);
+
+    return file.ListItemAllFields;
+  },
+
+  /**
+  * Updates a list item with system update
+  * @param webUrl Web url
+  * @param requestUrl The base URL without query parameters, pointing to the specific list where the item resides. This URL should represent the list.
+  * @param id The id of the list item
+  * @param properties An object of the properties that should be updated
+  * @param contentTypeName The name of the content type to update
+  * @param logger The logger object
+  * @param verbose If in verbose mode
+  * @returns The updated list item object
+  */
+  async systemUpdateListItem(requestUrl: string, id: string, logger: Logger, verbose: boolean, properties?: object, contentTypeName?: string): Promise<ListItemInstance> {
+    if (verbose && logger) {
+      logger.logToStderr(`getting request digest for systemUpdate request`);
+    }
+
+    const parsedUrl = new URL(requestUrl);
+    const serverRelativeSiteUrl = requestUrl.match(new RegExp('/sites/[A-Za-z0-9-]*'))![0];
+    const webUrl = `${parsedUrl.protocol}//${parsedUrl.host}${serverRelativeSiteUrl}`;
+
+    if (verbose && logger) {
+      logger.logToStderr(`Getting list id...`);
+    }
+
+    const listRequestOptions: CliRequestOptions = {
+      url: `${requestUrl}?$select=Id`,
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    const list = await request.get<{ Id: string; }>(listRequestOptions);
+
+    const listId = list.Id;
+    const res = await spo.getRequestDigest(webUrl);
+
+    const formDigestValue = res.FormDigestValue;
+    const objectIdentity: string = await spo.requestObjectIdentity(webUrl, logger, verbose);
+
+    const requestBodyOptions: string[] = properties ? Object.keys(properties).map(key => `
+    <Method Name="ParseAndSetFieldValue" Id="1" ObjectPathId="147">
+      <Parameters>
+        <Parameter Type="String">${key}</Parameter>
+        <Parameter Type="String">${(<any>properties)[key].toString()}</Parameter>
+      </Parameters>
+    </Method>`) : [];
+
+    const additionalContentType: string = contentTypeName ? `
+    <Method Name="ParseAndSetFieldValue" Id="1" ObjectPathId="147">
```

Comment:
This is probably invalid. Each Id in a CSOM call must be unique, so we shouldn't hardcode it to `1`